test/test.js

```diff
@@ -0,0 +1,45 @@
+const nock = require('nock')
+const got = require('got')
+const App = require('/path/to/app')
+
+describe('app.js', function() {
+  let app
+
+  beforeEach(function() {
+    // set up stuff
+    app = new App({
+      id: 123, privateKey: 'secrets'
+    })
```

Comment:
I'm not sure if mocha runs tests in parallel. If it does, you might run into a racing condition where the tests below use the same `app` instance. To work around this you can set the `app` instance on the the test context, for which you would do

```
this.app = new App({...})
```

and then in the tests you can use `this.app`. But note that you need to define the callback with `function () { /* ... */ }` in order to get the context (not `() => { /* ... */ }`. Otherwise `this` is not bound to the the `it()` context.

Reply:
Super good point; however, according to https://mochajs.org/:

> Mocha tests run serially, allowing for flexible and accurate reporting, while mapping uncaught exceptions to the correct test cases.

Serially = Sequentially = Not in parallel! Also the existence of multiple libraries that allow you to run mocha tests in parallel support this.

So I think that means it's okay as it is?

Reply:
sounds like it 👍 If at some point you see weird behaviour like a test passing if you run only this one but fails when running all tests, than that usually means the racing condition I mentioned above.